src/k8s/pkg/k8s/setup/kubeconfigs.go

```diff
@@ -2,17 +2,26 @@ package setup
 
 import (
 	"context"
+	"encoding/base64"
 	"fmt"
+	"net"
 	"os"
+	"strconv"
 
-	"github.com/canonical/k8s/pkg/utils/cert"
-	"github.com/canonical/k8s/pkg/utils"
 	apiImpl "github.com/canonical/k8s/pkg/k8sd/api/impl"
+	"github.com/canonical/k8s/pkg/snap"
+	"github.com/canonical/k8s/pkg/utils"
+	"github.com/canonical/k8s/pkg/utils/cert"
 	"github.com/canonical/microcluster/state"
 )
 
+type ExtraKubeConfigArgs struct {
+	ApiServerIp   *string
+	ApiServerPort *int
+}
+
 // InitKubeconfigs generates the kubeconfig files that services use to communicate with the apiserver.
-func InitKubeconfigs(ctx context.Context, state *state.State, ca *cert.CertKeyPair) error {
+func InitKubeconfigs(ctx context.Context, state *state.State, ca *cert.CertKeyPair, overwrites *ExtraKubeConfigArgs) error {
```

Comment:
`ExtraArgs` could also be a struct, not a pointer? simplifies the if-else logic later on?